test/gorest-api/gorest-api.test.js

```diff
@@ -12,82 +12,178 @@ describe("GoRest Users API test (/users)", () =>{
+           
+      expect(createdUser).to.have.property("name", userToBeCreated.name);     
+      expect(createdUser).to.have.property("gender", userToBeCreated.gender);
+      expect(createdUser).to.have.property("email", userToBeCreated.email);
+      expect(createdUser).to.have.property("status", userToBeCreated.status);
+      expect(createdUser).to.have.property("id").that.is.a("number")       
+    })    
+
+    it("Can create multiple users in parallel (POST /users", async () => {
+      const userToBeCreated1 = {
+        "name":"User1", 
+        "gender":"male",
+        "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`, 
+        "status":"active"};
+
+      const userToBeCreated2 = {
+        "name":"User2", 
+        "gender":"male",
+        "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`, 
+        "status":"active"};
+
+      const userToBeCreated3 = {
+        "name":"User3", 
+        "gender":"male",
+        "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`, 
+        "status":"active"};
+         
+
+      const myUserArray = await Promise.all([
+        api.createUser(userToBeCreated1),
+        api.createUser(userToBeCreated2),
+        api.createUser(userToBeCreated3)
+      ])
       
-      expect(response.data).to.have.property("name", userToBeCreated.name);
-      expect(response.data).to.have.property("gender", userToBeCreated.gender);
-      expect(response.data).to.have.property("email", userToBeCreated.email);
-      expect(response.data).to.have.property("status", userToBeCreated.status)
+      myUserArray.forEach((user) => {
+        expect(user.data).to.have.property("email").that.is.a("string")
+      })   
     })   
+  })
+
+  describe("GoRest Get users list (GET /users)", () => {
+    it("Can get a list of users (GET /users", async () => {
+      const response = await api.getUsersList();      
 
-    it("Error message returned when creating a user as an authenticated user (POST /users", async () => {
+      expect(response.meta.pagination).to.be.an("object");
+      expect(response.data, "Expected data array to have 20 results").to.be.an("array").that.has.lengthOf(20);
+      expect(response.meta.pagination).to.have.property("page", 1)
+    })  
+
+    it("Can filter a list of users by email (GET /users?name=varma", async () => {
       const userToBeCreated = {
         "name":"Tenali Ramakrishna", 
         "gender":"male",
         "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`, 
-        "status":"active"
-      }
-
-      const err = await guard(async () => unAuthenticatedApi.createUser(userToBeCreated));
-      
-      console.log('\x1b[36m%s\x1b[0m', JSON.stringify(err, null, 2));   
-      
-      //expect(response.data).to.have.property("name", userToBeCreated.name)
+        "status":"active"};    
+  
+      const userCreatedResponse = await api.createUser(userToBeCreated);
```

Comment:
I would create at least one more test user, that wouldn't match the search query, to make sure it's not included in the result set.